### PR TITLE
feat(presets): Add @rsjf/ monorepo

### PR DESCRIPTION
**Pull request by hawkrives**
## Changes

I added a monorepo, `@rjsf/*core*`, for the react-json-schema-form packages.

## Context

I was going to put this in locally, but I realized that I should just contribute the new monorepo back to core!

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

I believe that this updates the documentation automatically.

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/monorepo.ts [hunk 1]
```diff
@@ -276,6 +276,7 @@ const patternGroups = {
   clarity: ['^@cds/', '^@clr/'],
   embroider: '^@embroider/',
   fullcalendar: '^@fullcalendar/',
+  rjsf: '^@rjsf/',
```
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:279]: why not adding via sourceUrl like most of the others?

Comment (hawkrives) [reply lib/config/presets/internal/monorepo.ts:279]: Only because I wasn't sure of the difference! I'm happy to change it over.

**Review by viceice — COMMENTED** [review]
please validate that those packages do have correctly published the repo url. probably test it on a repo with a package rule. otherwise LGTM

**Comment by hawkrives** [comment]
Closing as #22248 was merged. Sorry that I didn't manage to come back to this!

**Pull request closed without merging**